lib/blob.js

```diff
@@ -147,17 +240,164 @@ function Blob(options) {
+ *                                    // It must be omitted if it has been specified in the associated stored access policy.
+ *     read:              false,      // Read the content, properties, metadata or block list of a blob or of
+ *                                    // any blob in the container if the resourceType is a container.
+ *     add:               false,      // Add a block to an append blob or to any append blob if the resourceType is a container.
+ *     create:            false,      // Write a new blob, snapshot a blob, or copy a blob to a new blob. These operations can be done to any blob in the container
+ *                                    // if the resourceType is a container.
+ *     write:             false,      // Create or write content, properties, metadata, or block list. Snapshot or lease the blob. Resize the blob (page blob only).
+ *                                    // These operations can be done for every blob in the container if the resourceType is a container
+ *     delete:            false,      // Delete the blob or any blob in the container if the resourceType is a container.
+ *     list:              false,      // List blobs in the container.
+ *   },
+ *   cacheControl:        '...',      // The value of the Cache-Control response header to be returned. (optional)
+ *   contentDisposition:  '...',      // The value of the Content-Disposition response header to be returned. (optional)
+ *   contentEncoding:     '...',      // The value of the Content-Encoding response header to be returned. (optional)
+ *   contentLanguage:     '...',      // The value of the Content-Language response header to be returned. (optional)
+ *   contentType:         '...',      // The value of the Content-Type response header to be returned. (optional)
+ *   ipAddresses:         '...',      // An IP address or a range of IP addresses. (optional)
+ *   accessPolicy:        '...'       // Reference to stored access policy (optional)
+ *                                    // A GUID string
+ * }
+ * ```
+ * @returns {string} Shared-Access-Signature on string form.
+ *
+ */
+Blob.prototype.sas = function sas(container, blob, options){
+  // verify the required options
+  assert(options, "options is required");
+  assert(options.resourceType, 'options.resourceType is required');
+  assert(options.resourceType === 'b' || options.resourceType === 'c',
+    'The possible values for options.resourceType are `b` or `c`');
+  assert(options.permissions || options.accessPolicy, "options.permissions or options.accessPolicy must be specified");
+
+  // Check that we have credentials
+  if (!this.options.accountId ||
+    !this.options.accessKey) {
+    throw new Error("accountId and accessKey are required for SAS creation!");
+  }
+
+  // Construct query-string with required parameters
+  var query = {
+    sv:   SERVICE_VERSION,
+    sr:   options.resourceType,
+    spr:  'https'
+  }
+
+  if (options.expiry) {
```

Comment:
Based on Azure documentation, the 'expiry' is optional if it has been specified in an associated stored access policy, and blob storage has support for access policy: setContainerACL/getContainerACL. For Queue and Table are not yet implemented. 
In order to prevent this 'no way to revoke the SAS' I tested to see if the permissions or accessPolicy were set:
assert(options.permissions || options.accessPolicy, "options.permissions or options.accessPolicy must be specified");
If we add a flag, as Jonas said, I think the API will be more flexible.